src/base/optimise/owl_algodiff_generic.ml

```diff
@@ -202,6 +203,18 @@ module Make
     | Arr x -> Arr A.(repeat ?axis x reps)
     | _     -> failwith "error: AD.repeat"
 
+  let repeat_mul axis x reps =
```

Comment:
If my understanding is correct, this function is meant to `repeat` on multiple axes given a ndarray, right? If so, we should probably improve the API rather than adding another adhoc function. 

Please have a look at tile function, if we make the repeat function interface similar to tile, do you think this can solve the problem?

Reply:
If so, please open an issue to suggest improving repeat function API asap, I can help you on this function.

Reply:
Thanks! That is exactly what this function is meant to do. I've renamed it to `repeat_axes` and added a TODO comment.